BDArmory/Control/BDAirspeedControl.cs

```diff
@@ -648,37 +648,19 @@ public void RCSEngineControl(FlightCtrlState s)
             // Control engines perpendicular to longitudinal axis to act as RCS thrusters using FlightCtrlState s.pitch/s.yaw/s.yaw/s.X/s.Y/s.Z inputs
             // Call this last of all control methods so FlightCtrlState is finalized
             Vector3 rcsTranslation = s.X * right + s.Y * up + s.Z * forward;
-            Vector3 rcsRotation = (vessel.ReferenceTransform.up + (s.pitch * -vessel.ReferenceTransform.forward + s.yaw * vessel.ReferenceTransform.right));
             float vesselRad = vessel.GetRadius();
```

Comment:
Instead of using the vessel max radius for each dimension, it'd be better to only consider the off-axis sizes for each moment. I.e.,
```diff
diff --git a/BDArmory/Control/BDAirspeedControl.cs b/BDArmory/Control/BDAirspeedControl.cs
index 422184faf..4d98462ec 100644
--- a/BDArmory/Control/BDAirspeedControl.cs
+++ b/BDArmory/Control/BDAirspeedControl.cs
@@ -650,3 +650,7 @@ public void RCSEngineControl(FlightCtrlState s)
             Vector3 rcsTranslation = s.X * right + s.Y * up + s.Z * forward;
-            float vesselRad = vessel.GetRadius();
+            Vector3 vesselRad = new( // vesselSize is width, height, length
+                (vessel.vesselSize.y + vessel.vesselSize.z) / 4f, // Pitch: average of height and length, halved
+                (vessel.vesselSize.x + vessel.vesselSize.z) / 4f, // Yaw: average of width and length, halved
+                (vessel.vesselSize.x + vessel.vesselSize.y) / 4f // Roll: average of width and height, halved
+            );
             for (int i = 0; i < rcsEngines.Count; i++)
@@ -659,5 +663,5 @@ public void RCSEngineControl(FlightCtrlState s)
                 // RCS Rotation using Moments
-                float pitchMoment = s.pitch * Vector3.Dot(vessel.ReferenceTransform.right, Vector3.Cross(rcsEngines[i].transform.position - vessel.CoM, rcsEngines[i].thrustTransforms[0].forward)) / (vesselRad);
-                float yawMoment = s.yaw * Vector3.Dot(vessel.ReferenceTransform.forward, Vector3.Cross(rcsEngines[i].transform.position - vessel.CoM, rcsEngines[i].thrustTransforms[0].forward)) / (vesselRad);
-                float rollMoment = s.roll * Vector3.Dot(vessel.ReferenceTransform.up, Vector3.Cross(rcsEngines[i].transform.position - vessel.CoM, rcsEngines[i].thrustTransforms[0].forward)) / (vesselRad);
+                float pitchMoment = s.pitch * Vector3.Dot(vessel.ReferenceTransform.right, Vector3.Cross(rcsEngines[i].transform.position - vessel.CoM, rcsEngines[i].thrustTransforms[0].forward)) / vesselRad.x;
+                float yawMoment = s.yaw * Vector3.Dot(vessel.ReferenceTransform.forward, Vector3.Cross(rcsEngines[i].transform.position - vessel.CoM, rcsEngines[i].thrustTransforms[0].forward)) / vesselRad.y;
+                float rollMoment = s.roll * Vector3.Dot(vessel.ReferenceTransform.up, Vector3.Cross(rcsEngines[i].transform.position - vessel.CoM, rcsEngines[i].thrustTransforms[0].forward)) / vesselRad.z;
                 giveThrust += pitchMoment + yawMoment + rollMoment; // Modify any translation to allow rotation
```